datafusion/src/logical_plan/dfschema.rs

```diff
@@ -160,7 +160,13 @@ impl DFSchema {
                 // field to lookup is qualified.
                 // current field is qualified and not shared between relations, compare both
                 // qualifer and name.
-                (Some(q), Some(field_q)) => q == field_q && field.name() == name,
+                (Some(q), Some(field_q)) => {
+                    if field_q.contains('.') {
+                        field_q.ends_with(q) && field.name() == name
```

Comment:
Wouldn't this incorrectly return true when `q` is a strict suffix of the table name in `field_q`? It would be more robust to check whether `field_q` ends with `'.'` + `q` right?

Reply:
Ah yes, good catch. Will address.